app/src/main/java/com/example/tasky/feature_agenda/data/util/PhotoValidatorImpl.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.tasky.feature_agenda.data.util
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import androidx.core.net.toUri
+import com.example.tasky.feature_agenda.domain.util.PhotoValidator
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+
+class PhotoValidatorImpl(
+    private val context: Context
+): PhotoValidator {
+
+    override suspend fun readUriAsByteArray(imageUriString: String): ByteArray? {
+        return withContext(Dispatchers.IO) {
+            val imageUri = imageUriString.toUri()
+            val imageBytes = context.contentResolver.openInputStream(imageUri)?.use {
+                it.readBytes()
+            }
+            imageBytes
+        }
+    }
+
+    override suspend fun saveImageBytes(imageBytes: ByteArray, parentDirectory: File, fileName: String) {
+        withContext(Dispatchers.IO) {
+            val file = File(parentDirectory, fileName)
+            FileOutputStream(file).use {
+                it.write(imageBytes)
+            }
+        }
+    }
```

Comment:
You don't need to save the photo locally. Just keep them in RAM until they're uploaded